lib/deas/router.rb

```diff
@@ -109,9 +135,37 @@ def apply_definitions!
       true
     end
 
+    def validate_trailing_slashes!
+      if self.trailing_slashes == REQUIRE_TRAILING_SLASHES
+        paths = []
+        all_have = self.routes.inject(true) do |result, route|
+          paths << route.path if route.path[-1..-1] != SLASH
+          result && route.path[-1..-1] == SLASH
```

Comment:
@kellyredding - So I hadn't thought about this but we might need special handling for urls with extensions? Particularly for us we use `.json` and `.html` routes and sometimes we use `.jpg` or some image format urls that don't just serve a static file and have to access the deas app.

Unfortunately, I think the general user would expect any routes with an extension on the end to ignore this directive. I don't think this issue relates for the require no trailing slashes directive and for the allow I think again you would want it to not retry a route with an extension.

Reply:
@jcredding great point and catch there.  Thanks man.  I'm going to address this, probably with some sort of `File.extname` gate-keeper logic, in a follow up effort.  I want this special handling to be in its own commit for reference purposes.

Reply:
@jcredding adding this for historical sake: we decided not to address this edge case and instead chose to remove the `require_trailing_slashes` directive.  See 4950917 for reference